modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -368,9 +362,12 @@ private <R> CompletableFuture<R> enlistInTx(
         return postEnlist(fut, false, actualTx, actualTx.implicit()).handle((r, e) -> {
             if (e != null) {
                 if (actualTx.implicit()) {
+                    // TODO: IGNITE-24244
+                    long timeout = actualTx.isReadOnly() ? actualTx.timeout() : 10_000;
```

Comment:
Let's introduce a temporary constant for 10000

Reply:
done